### PR TITLE
interop validation additions

### PR DESCRIPTION
**Pull request by jweissm**

### CODE REVIEW

File: stix2validator/validator.py [hunk 1]
```diff
@@ -758,7 +760,10 @@ def _schema_validate(obj, options, bundle_version=None):
```
Comment (clenk) [inline stix2validator/validator.py:760]:
```suggestion

    # Get validator for interop schemas if needed
    if options.interop and version > '2.0':
        interop_schema_dir = os.path.abspath(os.path.dirname(__file__) + '/schemas-'
                                             + version + '/interop/')
        interop_errors = _get_error_generator(obj['type'], obj, interop_schema_dir, version, default=core_schema)
        if interop_errors:
            error_gens.append((interop_errors, error_prefix))

```

File: stix2validator/v21/interop.py [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+            if 'created_by_ref' in obj:
+                rel_references.add(obj['id'])
+            else:
+                bad_references.add(obj['id'])
+
+    """Check if id is present in producers or normal identity SDOs"""
+    for obj in instance['objects']:
+        if obj['type'] != 'identity':
+            if 'created_by_ref' not in obj:
+                yield JSONError("Created by ref is not present in Object", obj['id'])
```
Comment (clenk) [inline stix2validator/v21/interop.py:25]:
```suggestion
                yield JSONError("created_by_ref is not present in Object", obj['id'])
```

File: stix2validator/validator.py [hunk 3]
```diff
@@ -623,7 +625,7 @@ def _get_error_generator(type, obj, schema_dir=None, version=DEFAULT_VER, defaul
     if schema_dir is None:
         default_path = True
         schema_dir = os.path.abspath(os.path.dirname(__file__) + '/schemas-'
-                                     + version + '/')
+                                     + version + '/schemas')
```
Comment (clenk) [inline stix2validator/validator.py:628]:
```suggestion
                                     + version + '/schemas/')
```

**Comment by clenk** [comment]
LGTM assuming https://github.com/oasis-open/cti-stix2-json-schemas/pull/146 gets merged and the schema submodules in this repo are updated.

**Comment by clenk** [comment]
@ejratl @sparrell This is ready for review/merging.

**Comment by ejratl** [comment]
Thanks for the nudge, @clenk and thanks for the contribution and the updates, @jweissm

**Pull request merged**